satMutMPRA.pm

```diff
@@ -0,0 +1,285 @@
+                  - other conditions: PKLR-48h, ZRSh-13h2, TERT-GAa, TERT-GBM, TERG-GSc
+
+The Bio::DB::HTS perl library or tabix utility must be installed in your path
+to use this plugin. The satMutMPRA data file can be downloaded from
+https://mpra.gs.washington.edu/satMutMPRA/
+
+satMutMPRA data can be downloaded for both GRCh38 and GRCh37 from the web server (https://mpra.gs.washington.edu/satMutMPRA/):
+'Download' section, select 'GRCh37' or 'GRCh38' for 'Genome release' and 'Download All Elements'.
+
+The file must be processed and indexed by tabix before use by this plugin.
+
+# GRCh38
+ > (grep ^Chr GRCh38_ALL.tsv; grep -v ^Chr GRCh38_ALL.tsv | sort -k1,1 -k2,2n ) | bgzip > satMutMPRA_GRCh38_ALL.gz
+ > tabix -s 1 -b 2 -e 2 -c C satMutMPRA_GRCh38_ALL.gz
+
+# GRCh37
+ > (grep ^Chr GRCh37_ALL.tsv; grep -v ^Chr GRCh37_ALL.tsv | sort -k1,1 -k2,2n ) | bgzip > satMutMPRA_GRCh37_ALL.gz
+ > tabix -s 1 -b 2 -e 2 -c C satMutMPRA_GRCh37_ALL.gz
+
+
+When running the plugin by default 'Value', 'P-Value', and 'Element'
+information is returned e.g.
+
+--plugin satMutMPRA,file=/path/to/satMutMPRA_GRCh38_ALL.gz
+
+You may include all columns with ALL; this fetches all data per variant
+(e.g. Tags, DNA, RNA, Value, P-Value, Element):
+
+--plugin satMutMPRA,file=/path/to/satMutMPRA_GRCh38_ALL.gz,cols=ALL
+
+You may want to select only a specific subset of information to be
+reported, you can do this by specifying the specific columns as parameters to the plugin e.g.
+
+--plugin satMutMPRA,file=/path/to/satMutMPRA_GRCh38_ALL.gz,cols=Tags:DNA
+
+If a requested column is not found, the error message will report the
+complete list of available columns in the satMutMPRA file. For a detailed description
+of the available information please refer to the manuscript or online web server.
+
+Tabix also allows the data file to be hosted on a remote server. This plugin is
+fully compatible with such a setup - simply use the URL of the remote file:
+
+--plugin satMutMPRA,file=http://my.files.com/satMutMPRA.gz
+
+Note that gene locations referred to in satMutMPRA may be out of sync with
+those in the latest release of Ensembl; this may lead to discrepancies with
+information retrieved from other sources.
+
+=cut
+
+package satMutMPRA;
+
+use strict;
+use warnings;
+
+use Bio::EnsEMBL::Variation::Utils::BaseVepTabixPlugin;
+
+use base qw(Bio::EnsEMBL::Variation::Utils::BaseVepTabixPlugin);
+
+my @fields_order;
+
+my $out_txt = 1;
+my $out_vcf = 0;
+my $out_json = 0;
+my $char_sep = "|";
+
+my %repl = (
+"LDLR.2"=>1, "SORT1.2"=>1,
+"SORT1-flip" =>1,
+"PKLR-48h" => 1, "ZRSh-13h2" => 1,
+"TERT-GAa" =>1, "TERT-GBM" =>1, "TERT-GSc" => 1);
+
+# default config
+my %DEFAULTS = (
+  pvalue => 0.00001,
+  incl_repl => 0,
+);
+
+sub new {
+  my $class = shift;
+
+  my $self = $class->SUPER::new(@_);
+
+  $self->expand_left(0);
+  $self->expand_right(0);
+
+  my $params_hash = $self->params_to_hash();
+  $DEFAULTS{$_} = $params_hash->{$_} for keys %$params_hash;
```

Comment:
Not a big deal, but it's a little confusing that the hash is called defaults even when the defaults are overwitten. It would be better called something like config.

Reply:
Good point, have changed the name. Thanks!